server/src/main/java/org/elasticsearch/internal/CompletionsPostingsFormatExtension.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.internal;
+
+import org.apache.lucene.search.suggest.document.CompletionPostingsFormat;
+
+/**
+ * Allows plugging-in the Completions Postings Format.
+ */
+public interface CompletionsPostingsFormatExtension {
```

Comment:
Maybe overkill also, but what happens if different nodes return a different codec name? Is it overkill to test that perhaps because we are not opening this extension to plugin developers?

Reply:
> what happens if that postings format is not found 

`CompletionFieldMapperTests#testPostingsFormat` already checks this [here](https://github.com/elastic/elasticsearch/blob/32dde26e4960b39a3a685cd617a18f9251533fb2/server/src/test/java/org/elasticsearch/index/mapper/CompletionFieldMapperTests.java#L154).


WRT " if we can verify that that works":  This is done via tests in the companion PR in the serverless repo.

Reply:
> Maybe overkill also, but what happens if different nodes return a different codec name? Is it overkill to test that perhaps because we are not opening this extension to plugin developers?

I'm not sure I follow, how would that happen?  That seems to be the purview of the extension itself, and not the mechanism that allows for it (the extension point) added in this PR.  WDYT?